ast/src/parsed/asm.rs

```diff
@@ -97,7 +81,7 @@ pub enum SymbolValueRef<'a> {
     /// A module definition
     Module(ModuleRef<'a>),
     /// A generic symbol / function.
-    Expression(&'a TypedExpression),
+    Expression(&'a Option<Expression>, &'a Option<TypeScheme<Expression>>),
```

Comment:
Why not keep TypedExpression? Lifetime issues?

Reply:
Because it's constructed from a reference over `PilStatement::LetStatement` which is
```
    LetStatement(
        SourceRef,
        String,
        Option<TypeScheme<Expression>>,
        Option<Expression>,
    ),
```
My understanding is that `SymbolValueRef::Expression` also covers statements like `let a: Type;` which seem to be allowed. `TypedExpression` would not cover that.